src/realm/list.cpp

```diff
@@ -325,6 +325,46 @@ Mixed ConstLstIf<T>::avg(size_t* return_cnt) const
     return AverageHelper<T>::eval(*m_tree, return_cnt);
 }
 
+template <class T>
+void ConstLstIf<T>::sort(std::vector<size_t>& indices, bool ascending) const
+{
+    auto sz = size();
+    auto sz2 = indices.size();
+
+    indices.reserve(sz);
+    if (sz < sz2) {
+        // If list size has decreased, we have to start all over
+        indices.clear();
+        sz2 = 0;
+    }
+    for (size_t i = sz2; i < sz; i++) {
```

Comment:
What if it has first decreased, then more has been added?

Reply:
The `indices` array should contain all indices relevant for the list. If the list has effectively grown 2 in size, 2 more indices are needed. When the list is shrinked I find that it is easier just to rebuild the array.